### ISSUE TITLE
Issue with credentials

### ISSUE THREAD

**Issue by antoniovittozzi** [ollo69/ha-smartthinq-sensors#723]
**Describe the bug**
I can't get integration works. Credentials are OK, LG Account is OK.

**Environment details:**
 - Environment (HASSIO, Raspbian, etc): HAOS Synology 920
 - Home Assistant version installed: Core 2024.3.1, Supervisor 2024.03.0,  Operating System 12.1, Frontend 20240307.0
 - Component version installed:
 - Last know working version: 
 - LG device type and model with issue:
 - LG devices connected (list):

**Output of HA logs**
2024-03-21 19:05:12.905 ERROR (MainThread) [custom_components.smartthinq_sensors] Error retrieving OAuth info from ThinQ
Traceback (most recent call last):
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 135, in get_oauth_info_from_login
    return await ClientAsync.oauth_info_from_user_login(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 1623, in oauth_info_from_user_login
    result = await Auth.oauth_info_from_user_login(username, password, gateway)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 910, in oauth_info_from_user_login
    token_info = await gateway.core.auth_user_login(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 540, in auth_user_login
    raise exc.AuthenticationError(msg)
custom_components.smartthinq_sensors.wideq.core_exceptions.AuthenticationError: code: MS.001.02 - message: Wrong Password:1
2024-03-21 19:06:16.309 WARNING (MainThread) [custom_components.smartthinq_sensors] Connection not available. ThinQ platform not ready
Traceback (most recent call last):
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 252, in async_setup_entry
    client = await lge_auth.create_client_from_token(refresh_token, oauth2_url)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 151, in create_client_from_token
    return await ClientAsync.from_token(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 1570, in from_token
    await client.refresh()
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 1488, in refresh
    self._auth = await self.auth.refresh(True)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 983, in refresh
    access_token, token_validity = await self._gateway.core.refresh_auth(
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 701, in refresh_auth
    out = await self._auth_request(
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 647, in _auth_request
    oauth_url = await self.get_oauth_url()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 430, in get_oauth_url
    async with self._get_session().post(
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 1194, in __aenter__
    self._resp = await self._coro
                 ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 504, in _request
    with timer:
  File "/usr/local/lib/python3.12/site-packages/aiohttp/helpers.py", line 735, in __exit__
    raise asyncio.TimeoutError from None
TimeoutError

**Additional context**

**Comment by kalmma** [comment]
I have the same problem

**Comment by erkaz** [comment]
I am having the same issue. I did recently change my password but it wasn't working prior and its been a few days since, still not loading. Here is the log from HA. 
Connection not available. ThinQ platform not ready
Traceback (most recent call last):
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 269, in async_setup_entry
    client = await lge_auth.create_client_from_token(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 153, in create_client_from_token
    return await ClientAsync.from_token(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 1593, in from_token
    await client.refresh()
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 1501, in refresh
    self._auth = await self.auth.refresh(True)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 989, in refresh
    access_token, token_validity = await self._gateway.core.refresh_auth(
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 707, in refresh_auth
    out = await self._auth_request(
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 653, in _auth_request
    oauth_url = await self.get_oauth_url()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 436, in get_oauth_url
    async with self._get_session().post(
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 1197, in __aenter__
    self._resp = await self._coro
                 ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 507, in _request
    with timer:
  File "/usr/local/lib/python3.12/site-packages/aiohttp/helpers.py", line 735, in __exit__
    raise asyncio.TimeoutError from None
TimeoutError

**Comment by juanchorossi** [comment]
Same over here!

**Comment by juanchorossi** [comment]
@antoniovittozzi 

After reading several related issues, I came across this one: https://github.com/ollo69/ha-smartthinq-sensors/issues/651, which mentions Pi-Hole blocking the traffic. This wasn't the case for me, but I noticed that the URL wasn't reachable even when using curl:

`https://ruic-service.lgthinq.com:46030/v1/service/application/dashboard`

I then discovered that, for some reason, my router (UDM Pro) was blocking outgoing traffic to that URL.

I modified the router settings, and it worked. Hope this helps!
